src/components/TypeComponents/SimpleSearch/Form/filters/SimpleSearchDateFilterField.js

```diff
@@ -29,7 +29,14 @@ const SimpleSearchDateFilterField = ({
   const { values } = useFormState();
   const { change } = useForm();
 
-  const isSetOrUnset = get(values, `${name}.comparator`) === 'isNull' || get(values, `${name}.comparator`) === 'isNotNull';
+  const comp = values?.[name?.comparator];
+  const comparatorSufficient =
+    comp === 'isNull' ||
+    comp === 'isNotNull' ||
+    comp === 'isSet' ||
+    comp === 'isNotSet' ||
+    comp === 'isEmpty' ||
+    comp === 'isNotEmpty';
```

Comment:
I suggest we have this logic as a function (so that we follow the DRY principle) as its being repeated across multiple files. So you could create a `utilities` folder probably at the `TypeComponents` folder level and add a utility function named `isComparatorPresent`

and inside `isComparatorPresent.js`

 ```
   const comparatorList = ['isNull', 'isNotNull', .....]; 
   export default function isComparatorPresent(comparator) {
        return comparatorList.includes(comparator)
   }
```

You can refer to the [utilities](https://github.com/folio-org/ui-agreements/tree/master/src/components/utilities) folder here to get an idea as to how to export and organize the functions.